src/dev/build/tasks/nodejs_modules/code_parser/visitors.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function dependenciesVisitorsGenerator(dependenciesAcc) {
+  return (() => {
+    // This was built based on two main tools: an ast explorer and the
+    // main docs for the Esprima
+    //
+    // https://astexplorer.net
+    // https://esprima.readthedocs.io/en/latest/syntax-tree-format.html
```

Comment:
You may need to add `@notice` for the licenses for these projects, even if we don't depend on them

Reply:
added 👍